src/Octoshift/Services/AdoApi.cs

```diff
@@ -700,6 +736,40 @@ public virtual async Task<bool> IsCallerOrgAdmin(string org)
         return await HasPermission(org, collectionSecurityNamespaceId, genericWritePermissionBitMaskValue);
     }
 
+    private string ExtractErrorMessage(string response, string dataProviderKey)
+    {
+        if (!response.HasValue())
+        {
+            return null;
+        }
+
+        try
+        {
+            var data = JObject.Parse(response);
+            if (data["dataProviders"] is not JObject dataProviders)
+            {
+                return null;
+            }
+
+            return dataProviders[dataProviderKey] is not JObject dataProvider ? null : (string)dataProvider["errorMessage"];
+        }
+        catch (JsonException)
+        {
+            // Return null for any JSON parsing errors to gracefully handle malformed responses
+            return null;
+        }
+        catch (ArgumentException)
+        {
+            // Return null for any JSON parsing errors to gracefully handle malformed responses
+            return null;
+        }
+        catch (InvalidOperationException)
+        {
+            // Return null for any JSON parsing errors to gracefully handle malformed responses
+            return null;
+        }
```

Comment:
While this is "graceful" it is not user friendly. Don't swallow the errors unreported.